modules/core/src/test/scala/sangria/federation/FederationSpec.scala

```diff
@@ -0,0 +1,136 @@
+package sangria.federation
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import sangria.util.FutureResultSupport
+import sangria.macros._
+import sangria.execution.{ExecutionScheme, Executor}
+import sangria.marshalling.InputUnmarshaller
+import sangria.parser.QueryParser
+import sangria.schema.{AdditionalTypes, Argument, AstSchemaBuilder, FieldResolver, Schema}
+import sangria.validation.Violation
+
+import scala.util.Success
+
+class FederationSpec extends AnyWordSpec with Matchers with FutureResultSupport {
+  import FederationSpec._
+
+  "To support Apollo Federation, an unmarshaller" should {
+    "accept a map value as a scalar" in {
+      val AST =
+        graphql"""
+           schema {
+             query: Query
+           }
+
+           type Query {
+             entities(representation: _Any!): _Any
+           }
+
+            union _Entity = State
+
+            type State {
+             id: Int!
+             value: String!
+           }
+         """
+
+      val federationSupport =
+        new FederationSupport(sangria.marshalling.json4s.jackson.Json4sJacksonInputUnmarshaller)({
+          case obj: org.json4s.JObject => obj
+        })
+
+      val schema: Schema[Any, Any] = Schema.buildFromAst(
+        AST,
+        AstSchemaBuilder.resolverBased[Any](
+          FieldResolver.map(
+            "Query" -> Map(
+              "entities" -> (ctx => ctx.withArgs(federationSupport.representationArg)(arg => arg))
+            )
+          ),
+          AdditionalTypes(federationSupport._Any.Type)
+        )
+      )
```

Comment:
@xsoufiane I'm wondering if we could put those lines into `FederationSupport`. With schema extensions maybe?

Reply:
Great idea ❤️ I will spend some time trying to figure out if we can do it generically for all the other marshallers